solver/llbsolver/bridge.go

```diff
@@ -97,14 +99,16 @@ func (b *llbBridge) Solve(ctx context.Context, req frontend.SolveRequest) (res s
 		}
 	}
 
-	if res != nil {
-		wr, ok := res.Sys().(*worker.WorkerRef)
-		if !ok {
-			return nil, nil, errors.Errorf("invalid reference for exporting: %T", res.Sys())
-		}
-		if wr.ImmutableRef != nil {
-			if err := wr.ImmutableRef.Finalize(ctx); err != nil {
-				return nil, nil, err
+	for _, r := range res {
+		if r != nil {
+			wr, ok := r.Sys().(*worker.WorkerRef)
+			if !ok {
+				return nil, nil, errors.Errorf("invalid reference for exporting: %T", r.Sys())
```

Comment:
Maybe add `r.ID()` in error message for easier debug?

Reply:
If it is not a `WorkerRef` then how would you use that key for debugging. Only use for ID is if it is a `WorkerRef` key.